bokeh/models/annotations.py

```diff
@@ -28,8 +28,23 @@ class Legend(Annotation):
 
     """
 
-    orientation = Enum(Orientation, help="""
-    The location where the legend should draw itself.
+    __deprecated_attributes__ = ('orientation',)
+
+    @property
+    @deprecated("Bokeh 0.11", "Legend.location")
+    def orientation(self):
+        return self.location
+
+    @orientation.setter
+    @deprecated("Bokeh 0.11", "Legend.location")
+    def orientation(self, location):
+        self.location = location
+
+    location = Either(Enum(LegendLocation), Tuple(Float, Float),
+        default="top_right", help="""
+    The location where the legend should draw itself. It's either one of
+    ``bokeh.enums.LegendLocation``'s enumerated values, or a ``(x, y)``
+    tuple indicating an absolute location (from the bottom-left corner).
```

Comment:
It would be trivial to use the default mappers to map data coordinates to screen coordinates, and if someone wants to orient a legend wrt some data locations it doesn't seem unreasonable (tho they'd probably want to disable panning/zooming in this case)